contractcourt/chain_arbitrator.go

```diff
@@ -1192,9 +1232,6 @@ func (c *ChainArbitrator) ForceCloseContract(chanPoint wire.OutPoint) (*wire.Msg
 // channel has finished its final funding flow, it should be registered with
 // the ChainArbitrator so we can properly react to any on-chain events.
 func (c *ChainArbitrator) WatchNewChannel(newChan *channeldb.OpenChannel) error {
-	c.Lock()
```

Comment:
Not sure but I think we cannot lock the `chainArb` here and also start the ChannelArbitrator, because the ChannelArbitrator might call the `ResolveContract` which needs the ChainArb lock as well, so probably this deadlock was never seen in the wild but I think we need to unlock the `ChainArb` before starting the ChannelArb ?